calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -0,0 +1,309 @@
+import requests
+import sh
+
+# Append to existing env, to avoid losing PATH etc.
+# Need to edit the path here since calicoctl loads client on import.
+# TODO-PAT: This shouldn't be hardcoded
+os.environ['ETCD_AUTHORITY'] = 'kubernetes-master:6666'
+
+from calicoctl import container_add
+from pycalico.datastore import IF_PREFIX
+from pycalico.util import generate_cali_interface_name
+
+CALICOCTL_PATH = os.environ.get('CALICOCTL_PATH', '/home/vagrant/calicoctl')
+calicoctl = sh.Command(CALICOCTL_PATH).bake(_env=os.environ)
+
+ETCD_AUTHORITY_ENV = "ETCD_AUTHORITY"
+PROFILE_LABEL = 'CALICO_PROFILE'
+ETCD_PROFILE_PATH = '/calico/'
+AllowRule = namedtuple('AllowRule', ['port', 'proto', 'source'])
+
+
+class NetworkPlugin(object):
+    def __init__(self):
+        self.pod_name = None
+        self.docker_id = None
+
+    def create(self, args):
+        """"Create a pod."""
+        # Calicoctl only
+        self.pod_name = args[3].replace('-', '_')
+        self.docker_id = args[4]
+
+        print('Configuring docker container %s' % self.docker_id)
+
+        try:
+            endpoint = self._configure_interface()
+            self._configure_profile(endpoint)
+        except CalledProcessError as e:
+            print('Error code %d creating pod networking: %s\n%s' % (
+                e.returncode, e.output, e))
+            sys.exit(1)
+
+    def delete(self, args):
+        """Cleanup after a pod."""
+        self.pod_name = args[3].replace('-', '_')
+        self.docker_id = args[4]
+
+        # Remove the profile for the workload.
+        calicoctl('container', 'remove', self.docker_id)
+        calicoctl('profile', 'remove', self.pod_name)
+
+    def _configure_interface(self):
+        """Configure the Calico interface for a pod."""
+        container_ip = self._read_docker_ip()
+        self._delete_docker_interface()
+        print('Configuring Calico networking.')
+        ep = container_add(self.docker_id, container_ip, 'eth0')
+        interface_name = generate_cali_interface_name(IF_PREFIX, ep.endpoint_id)
+        node_ip = self._get_node_ip()
+        print('Adding IP %s to interface %s' % (node_ip, interface_name))
+        check_call(['ip', 'addr', 'add', node_ip + '/32',
+                    'dev', interface_name])
+        print('Finished configuring network interface')
+        return ep
+
+    def _get_node_ip(self):
+        """Get the IP for the host node from the k8s API."""
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        s.connect(('8.8.8.8', 80))
+        ip = s.getsockname()[0]
+        s.close()
+        return ip
+
+    def _read_docker_ip(self):
+        """Get the IP for the pod's infra container."""
+        ip = check_output([
+            'docker', 'inspect', '-format', '{{ .NetworkSettings.IPAddress }}',
+            self.docker_id
+        ])
+        # Clean trailing whitespace (expect a '\n' at least).
+        ip = ip.strip()
+
+        print('Docker-assigned IP was %s' % ip)
```

Comment:
Logs probably want to include the docker ID / pod so we know which container / pod we're talking about.